bundles/org.openhab.binding.growatt/src/main/java/org/openhab/binding/growatt/internal/dto/GrottDevice.java

```diff
@@ -44,4 +51,22 @@ public String getDeviceId() {
     public @Nullable GrottValues getValues() {
         return values;
     }
+
+    /**
+     * Return the time stamp of the data DTO sent by the inverter data-logger.
+     * <p>
+     * Note: the inverter provides a time stamp formatted as a {@link LocalDateTime} without any time zone information,
+     * so we convert it to an {@link Instant} based on the OH system time zone. i.e. we are forced to assume the
+     * inverter and the OH PC are both physically in the same time zone.
+     *
+     * @return the time stamp {@link Instant}
+     */
+    public @Nullable Instant getTimeStamp() {
+        try {
+            return ZonedDateTime.of(LocalDateTime.parse(Objects.requireNonNull(timeStamp)), ZoneId.systemDefault())
```

Comment:
The inverter will 99.999% of the time be located in the same house as the OH server, so I don't think there is any real risk. Furthermore if the 0.001% case would occur, the newly added `inverter-clock-offset` channel will contain a value of +/- >> 3600 seconds.

Note: the `inverter-clock-offset` is a special (advanced) channel that I added because on the special days of changeover between summer and winter time, the inverter and server clocks can be out of synch, and that may cause some odd results in a night time low tariff charging program on that night.